alibi_detect/utils/pytorch/kernels.py

```diff
@@ -88,6 +168,271 @@ def forward(self, x: Union[np.ndarray, torch.Tensor], y: Union[np.ndarray, torch
         return kernel_mat.mean(dim=0)  # [Nx, Ny]
 
 
+class RationalQuadratic(BaseKernel):
+    def __init__(
+        self,
+        alpha: torch.Tensor = None,
+        init_fn_alpha: Callable = pseudo_init_fn,
+        sigma: torch.Tensor = None,
+        init_fn_sigma: Callable = sigma_median,
+        trainable: bool = False,
+        active_dims: Optional[list] = None,
+        feature_axis: int = -1
+    ) -> None:
+        """
+        Rational Quadratic kernel: k(x,y) = (1 + ||x-y||^2 / (2*sigma^2))^(-alpha).
+        A forward pass takesa batch of instances x [Nx, features] and y [Ny, features]
+        and returns the kernel matrix [Nx, Ny].
+
+        Parameters
+        ----------
+        alpha
+            Exponent parameter of the kernel.
+        sigma
+            Bandwidth used for the kernel.
+        """
+        super().__init__()
+        self.parameter_dict['alpha'] = 'exponent'
+        self.parameter_dict['sigma'] = 'bandwidth'
+        if alpha is None:
+            self.raw_alpha = nn.Parameter(torch.empty(1), requires_grad=trainable)
+            self.init_required = True
```

Comment:
At the moment it seems we would have the unexpected behaviour whereby if the user specified alpha but not sigma (or vice versa) then both get inferred from scratch?

Reply:
With the new kernel implementation, each parameter can individually specify whether initialisation is required. The pre-set values can then be kept.